c7n/actions.py

```diff
@@ -113,43 +113,100 @@ class ModifyGroupsAction(BaseAction):
     Note an interface always gets at least one security group, so
     we also allow specification of an isolation/quarantine group
     that can be specified if there would otherwise be no groups.
+
+    type: modify-security-groups
+        add: []
+        remove: [] | matched
+        isolation-group: sg-xyz
     """
 
+    properties = {
+        'add': {'oneOf': [{'type': 'string'}, {'type': 'array', 'items': 'string'}]},
+        'remove': {'oneOf': [
+            {'type': 'array', 'items': 'string'},
+            {'enum': ['matched', 'all', {'type': 'string'}]}]},
+        'isolation-group': {'oneOf': [{'type': 'string'}, {'type': 'array', 'items': 'string'}]}
+    }
     schema = utils.type_schema(
-        'remove-groups',
-        **{'groups': {'anyOf': [
-            {'type': 'string', 'enum': ['matched', 'all']},
-            {'type': 'array', 'items': {'type': 'string'}}]},
-           'isolation-group': {'type': 'string'}})
+        'modify-security-groups',
+        **properties
+    )
+
+    def validate(self):
+        if 'remove' in self.data:
```

Comment:
definitely could use a doc string on the validation/contract its trying to enforce.